homeassistant/components/wallbox/config_flow.py

```diff
@@ -47,14 +59,27 @@ async def async_step_user(self, user_input=None):
         errors = {}
 
         try:
-            info = await validate_input(self.hass, user_input)
+            await self.async_set_unique_id(user_input["station"])
+            if not self._reauth_entry:
+                self._abort_if_unique_id_configured()
+                info = await validate_input(self.hass, user_input)
+                return self.async_create_entry(title=info["title"], data=user_input)
+            if user_input["station"] == self._reauth_entry.data[CONF_STATION]:
+                self.hass.config_entries.async_update_entry(
+                    self._reauth_entry, data=user_input, unique_id=user_input["station"]
+                )
+                self.hass.async_create_task(
+                    self.hass.config_entries.async_reload(self._reauth_entry.entry_id)
+                )
+                return self.async_abort(reason="reauth_successful")
+            errors["base"] = "reauth_invalid"
         except ConnectionError:
             errors["base"] = "cannot_connect"
         except InvalidAuth:
             errors["base"] = "invalid_auth"
-        else:
-            return self.async_create_entry(title=info["title"], data=user_input)
 
         return self.async_show_form(
-            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+            step_id="user",
+            data_schema=STEP_USER_DATA_SCHEMA,
```

Comment:
Normally we should only ask the user for the required info to re-authenticate in the reauth flow. We shouldn't ask the user again for info that should stay the same like station id.